src/common/mfu_flist_walk.c

```diff
@@ -93,6 +93,35 @@ static void reduce_fini(const void* buf, size_t size)
     MFU_LOG(MFU_LOG_INFO, "Walked %llu items in %.3lf secs (%.3lf items/sec) ...", val, secs, rate);
 }
 
+/****************************************
+ * Global helper functions
+ ***************************************/
+
+/** Build a full path from a dirname and basename in the form:
+ * <dir> + '/' + <name> + '/0'
+ * up to path_len long.
+ * Returns 0 on success and -1 if the new path is too long. */
+static int build_path(char* path, size_t path_len, const char* dir, const char* name)
+{
+    size_t dir_len = strlen(dir);
+
+    /* Only separate with a '/' if the dir does not have a trailing slash.
+     * Builds a path to at most path_len long. */
+    int new_len;
+    if ((dir_len > 0) && (dir[dir_len - 1] == '/')) {
+        new_len = snprintf(path, path_len, "%s%s", dir, name);
+    } else {
+        new_len = snprintf(path, path_len, "%s/%s", dir, name);
+    }
+    if (new_len > path_len) {
+        MFU_LOG(MFU_LOG_ERR, "Path name is too long, %lu chars exceeds limit %lu: '%s/%s'",
+                new_len, path_len, dir, name);
```

Comment:
@daltonbohning do I have this right that this will remove an extra "/" if one is already after the directory name?

Reply:
@dsikich Right. Currently, this is what happens:
```
// Duplicate // in path
"/dirname/" + "/" + "filename" = "/dirname//filename"
```
But this PR changes that to:
```
// No duplicate //
"/dirname/" + "filename" = "/dirname/filename"
"/dirname" + "/" + "filename" = "/dirname/filename"
```

Realistically, this was only happening when dirname was "/" (root)